Make Popup annotations always have noRotate flag set as true

yyliu12 opened the pull request:
In PDF, pages can be rotated using their Rotate property (7.7.3.3). An annotation on a PDF can have a NoRotate flag set, in which case the annotation will not rotate (12.5.3). The PDF specification also requires that some annotations will always have the NoRotate flag set, including the Text annotation (12.5.6.4). 

The NoRotate flag was implemented in PDF.js in pull request #16110. However, the implementation only works if a PDF editor creates both a Popup annotation and a Text annotation. Adobe Acrobat does this. However, some PDF editors only create a Text annotation. Even if the Text annotation has the NoRotate flag set, the popup created by PDF.js will not have the NoRotate flag. This is because the annotation layer [does not pass the NoRotate flag](https://github.com/mozilla/pdf.js/blob/85b67f19bcfd2d5156d04cc69bc40930919a6f54/src/display/annotation_layer.js#L608) to the Popup object when creating it. This pull request fixes that. 

I have created a test PDF with a single page that has the Rotate property set. The left comment was created using Adobe Acrobat while the right comment was created using a PDF editor that does not create the Popup annotation. 

Here's how the right comment renders in Acrobat:

![image](https://github.com/user-attachments/assets/bde25d55-c3da-40b1-bdb6-d4062b32a707)

Here's how it displays in the version of pdf.js available at mozilla.github.io:

![image](https://github.com/user-attachments/assets/0f401bc5-bcc9-43bc-8129-22954a03bf89)

And here's how it displays in pdf.js after applying the code attached to this pull request:

![image](https://github.com/user-attachments/assets/ca96de6f-7004-4764-8be6-56c0fd50e636)

Test PDF: [Testing file.pdf](https://github.com/user-attachments/files/20952960/Testing.file.pdf)

calixteman commented on src/display/annotation_layer.js:
```diff
@@ -615,6 +615,7 @@ class AnnotationElement {
         borderStyle: 0,
         id: `popup_${data.id}`,
         rotation: data.rotation,
+        noRotate: data.noRotate,
```
I think we should always set `noRotate` to `true` because as far as I can tell it doesn't really make sense to have to break your neck in order to try to read a rotated popup.
@timvandermeij wdyt ?

calixteman replied:
As far as I know, popup aren't printed and if they're because they've an appearance stream we don't care about the rotation angle in the viewer itself. Am I missing something ?

calixteman commented:
@yyliu12 would you mind to update your patch in order to take into account: https://github.com/mozilla/pdf.js/pull/20043#discussion_r2172750330 ?
Thank you.

timvandermeij commented:
In addition to the above, please also add a regression test for this behavior. You can add the file you already created to `test/pdfs` as `pr20043.pdf` and then use the `add_test.mjs` tool (here: https://github.com/mozilla/pdf.js/blob/master/test/add_test.mjs) to update the manifest file.

yyliu12 committed:
Make Popup annotations always have noRotate flag set as true

Necessary because when there is no Popup annotation created along
with a Text annotation, the Popup annotation created by pdf.js
does not receive the noRotate flag

calixteman commented:
/botio test

timvandermeij reviewed (approved):
The change looks good to me now, but before we merge this please put the nice context from the PR description (in a summarized way) in the commit message itself so it's also visible using e.g. `git log` for future reference.

Thank you for your contribution!

calixteman commented:
/botio makeref
